### PR TITLE
chore: hoist CreateContentMenu into react-forms

### PR DESCRIPTION
**Pull request by jbevis**
This PR will refactor CreateContentMenu.tsx to be a component inside of ```react-forms``` instead of duplicated inside of ```react-sidebar``` and ```react-toolbar```

### CODE REVIEW

**Comment by spbyrne** [comment]
Hey, looks good to me. I'm going to do a build & test locally before approving.

**Comment by spbyrne** [comment]
There's a slight UI issue I'll fix. The dropdown menu opens in the wrong direction now for the sidebar.

**Review by spbyrne — DISMISSED** [review]
Looks good 👍

**Review by dwalkr — COMMENTED** [review]
Looks good. Noticed there aren't any dependency changes - `@tinacms/react-forms` is a peer dep of both the sidebar & toolbar packages. I'm wondering if it should now be a direct dependency.

**Comment by DirtyF** [comment]
Given I run the next demo locally 
When I add a new post from /blog/apple and click on Create
Then I get a `rejected` error message and no console output

CleanShot 2021-05-26 at 15 55 08@2x

File: packages/demo-next/pages/blog/[slug].tsx [hunk 1]
```diff
@@ -38,6 +38,13 @@ const addBlogPlugin: ContentCreatorPlugin<any> = {
     {
       name: 'title',
       component: 'text',
+      validate: value => {
```
Comment (Enigmatical) [inline packages/demo-next/pages/blog/[slug].tsx:41]: I'm not sure we want to commit these changes; looks like they are strictly the result of debugging.

**Review by Enigmatical — CHANGES_REQUESTED** [review]
Rest of the PR looks good; barring the comments from @dwalkr.

File: packages/demo-next/pages/blog/[slug].tsx (hunk 1 above)
Comment (DirtyF) [reply packages/demo-next/pages/blog/[slug].tsx:41]: Down with Chris, either it works for the demo prupose or we don't ship it.

**Comment by jbevis** [comment]
@dwalkr so are you suggesting that ```@tinacms/react-forms``` be removed from ```peerDependencies``` in both the sidebar and toolbar packages, and then added as a dependency as well or just remove it as ```@tinacms/react-forms``` is also a devDependency for both. 

The clarification is more for my own understanding as to how the all the packages within the monorepo are wired together.

File: packages/demo-next/pages/blog/[slug].tsx (hunk 1 above)
Comment (jbevis) [reply packages/demo-next/pages/blog/[slug].tsx:41]: Agreed, I'll prune the debugging changes from the next demo.

**Review by dwalkr — DISMISSED** [review]

**Review by dwalkr — APPROVED** [review]

File: packages/demo-next/pages/blog/[slug].tsx (hunk 1 above)
Comment (jbevis) [reply packages/demo-next/pages/blog/[slug].tsx:41]: @Enigmatical those changes have been pruned out.

**Comment by jbevis** [comment]
I'll be honest, this PR has gotten a bit messy...

The new lock file also has me feeling a bit queasy. I'm going to close this to clean up the branch locally and then try out a new PR hopefully with less issues.

**Pull request closed without merging**